src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1002,6 +1026,211 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
+                                                                                  const otSrpClientService * aRemovedServices,
+                                                                                  void * aContext)
+{
+    switch (aError)
+    {
+    case OT_ERROR_NONE: {
+        ChipLogProgress(DeviceLayer, "OnSrpClientNotification: Last requested operation completed successfully");
+
+        if (aRemovedServices)
+        {
+            otSrpClientService * service = const_cast<otSrpClientService *>(aRemovedServices);
+            otSrpClientService * next    = nullptr;
+
+            // Free memory for all removed services.
+            do
+            {
+                next = service->mNext;
+                // Clear also names associated with the SrpClient::Service object
+                service->mInstanceName[0] = '\0';
+                service->mName[0]         = '\0';
+                memset(service, 0, sizeof(otSrpClientService));
+                service = next;
+            } while (service);
+        }
+        break;
+    }
+    case OT_ERROR_PARSE:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Parsing operaton failed");
+        break;
+    case OT_ERROR_NOT_FOUND:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Domain name or RRset does not exist");
+        break;
+    case OT_ERROR_NOT_IMPLEMENTED:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Server does not support query type");
+        break;
+    case OT_ERROR_SECURITY:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Operation refused for security reasons");
+        break;
+    case OT_ERROR_DUPLICATED:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Domain name or RRset is duplicated");
+        break;
+    case OT_ERROR_RESPONSE_TIMEOUT:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Timed out waiting on server response");
+        break;
+    case OT_ERROR_INVALID_ARGS:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Invalid service structure detected");
+        break;
+    case OT_ERROR_NO_BUFS:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Insufficient buffer to handle message");
+        break;
+    case OT_ERROR_FAILED:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Internal server error occurred");
+        break;
+    default:
+        ChipLogError(DeviceLayer, "OnSrpClientNotification: Unknown error occurred");
+        break;
+    }
+}
+
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientStateChange(const otSockAddr * aServerSockAddr,
+                                                                                 void * aContext)
+{
+    if (aServerSockAddr)
+    {
+        ChipLogProgress(DeviceLayer,
+                        "OnSrpClientStateChange: SRP Client was started, as detected server addressed: %x:%x:%x:%x:%x:%x:%x:%x",
```

Comment:
Does such long message print correctly? I think Zephyr has some limits so it's worth checking or shorten the message (e.g. by removing the OnSrpClientStateChange: prefix).

Reply:
Yeah it prints correctly, however I shortened it a little just in case.